app/ui/FormControlNumber.tsx

```diff
@@ -0,0 +1,33 @@
+import { ReactElement } from "react";
+import { Form, FormControlProps } from "react-bootstrap";
+import { greyButtonBackground } from "./colourLibrary";
+
+export function FormControlNumber(
+  props: Omit<FormControlProps, "onChange"> & {
+    onChange: (newInput: string) => void;
+    width?: number;
+  }
+): ReactElement {
+  const { width, onChange, ...otherProps } = props;
+  return (
+    <Form.Control
+      {...otherProps}
+      onChange={(e) => {
+        const value = e.target.value;
+        const valueAsNumber = Number(value);
+
+        if (isNaN(valueAsNumber)) {
+          return;
+        }
+        onChange(value);
+      }}
+      style={{
+        width: width !== undefined ? width : 160,
```

Comment:
using your new skills in reading logical statements, you might be able to write this a bit more concisely in the next PR